File: src/main/java/de/weimarnetz/registrator/services/NetworkVerificationService.java
package de.weimarnetz.registrator.services;

import org.springframework.stereotype.Component;

import de.weimarnetz.registrator.configuration.NetworksConfiguration;

import lombok.AllArgsConstructor;


@Component
@AllArgsConstructor
public class NetworkVerificationService {

    private final NetworksConfiguration networksConfiguration;

    public boolean isNetworkValid(String network) {
        return networksConfiguration.getMap().containsKey(network);
    }
}

File: src/main/java/de/weimarnetz/registrator/services/NetworkVerificationService.kt
package de.weimarnetz.registrator.services

import de.weimarnetz.registrator.configuration.NetworksConfiguration
import org.springframework.stereotype.Component

@Component
class NetworkVerificationService(private val networksConfiguration: NetworksConfiguration) {
    fun isNetworkValid(network: String?): Boolean {
        return networksConfiguration.map.containsKey(network)
    }
}